CMakeLists.txt

```diff
@@ -48,8 +48,8 @@ ENDIF()
 FIND_PROGRAM(CMAKE_AR_TMP NAMES ar)
 
 # detect top level projects
-FIND_PACKAGE( Yajl )
```

Comment:
Would the following meet your needs?:
IF(USE_SYSTEM_YAJL)
  FIND_PACKAGE(Yajl REQUIRED)
ELSE()
  FIND_PACKAGE( Yajl )
ENDIF()
IF(YAJL_FOUND)
  ... as is ...

The reason is, the config stage shall not fail if user give no -DUSE_SYSTEM_YAJL. I consider it optional.